### ISSUE TITLE
How to train the network to obtain class accuracy for CMU-MOSIE sentiment data ?

### ISSUE THREAD

**Issue by shamanez** [yaohungt/Multimodal-Transformer#17]
What is the loss function used to obtain seven class accuracy? Is it softmax?

**Comment by yaohungt** [comment]
No, we follow the prior work, which used MAE as the loss function between the model's output and the ground truth sentiment score. Then, we round the output to integers and perform classification.

This process was how it did in prior work.

**Comment by shamanez** [comment]
This part is not in your code right?

**Comment by yaohungt** [comment]
It is in the file eval_metrics.py

https://github.com/yaohungt/Multimodal-Transformer/blob/master/src/eval_metrics.py

**Comment by shamanez** [comment]
ok found it. I also would like to know about the loss function. You guys have used negative log-likelihood for binary classification right?

**Comment by shamanez** [comment]
According to the CMU-Multimodal SDK, there are 7 classes for the sentiment. 
https://github.com/A2Zadeh/CMU-MultimodalSDK/tree/master/mmsdk/mmdatasdk/dataset/standard_datasets/CMU_MOSEI

**Comment by yaohungt** [comment]
No, it is the same as the multiclass one. We use rounding and then perform the binary classification. 

In short, these procedures may not be optimal, but they are strictly following the prior work. 

I think it should be fine if you use any other metric or loss as long as you clarify the differences.

**Comment by yaohungt** [comment]
I am not the author of this SDK, but at the moment when I download the data from it, the label is provided as a scalar value ranges [-3,3].

**Comment by shamanez** [comment]
yeah. In the SDK they show how to categorize that score (-3,+3) into 7 classes. So I am thinking of running your model with a softmax, Is it ok?

**Comment by yaohungt** [comment]
Yes, it is OK. I have tried it, and the performance will not differ too much from using MAE as a loss and rounding to classification. I can confirm this.

**Comment by shamanez** [comment]
Thanks a lot for your quick help. This means a lot to me. :)
